### PR TITLE
feat(tags): Add environment_id to tagstore create APIs

### PR DESCRIPTION
**Pull request by bretthoerner**
This is *just* dealing with the create APIs. Note that the legacy
backend ignores the environment that is passed in, but I've made sure to
pass in a valid environment_id everywhere these methods are called.

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/tasks/unmerge.py [hunk 1]
```diff
@@ -267,7 +267,8 @@ def collect_tag_data(events):
     results = {}
 
     for event in events:
-        tags = results.setdefault(event.group_id, {})
+        environment = get_environment(event)
```
Comment (tkaemming) [inline src/sentry/tasks/unmerge.py]: This was renamed as part of GH-6492.

**Review by tkaemming — CHANGES_REQUESTED** [review]
This generally looks good but your build is broken (see inline comment.)

Are you planning on filling the legacy backend, or does that depend on what we decide the future plan is?

**Comment by bretthoerner** [comment]
When you say filling do you mean "making use of `environment_id`"? If so, no, `legacy` will always ignore it. The new backend will have the columns and be able to support it.

**Comment by tkaemming** [comment]
That answers my question. I accidentally a word in my comment above (I meant to write "are you planning on filling out" which I guess wasn't super clear to begin with.)

**Comment by bretthoerner** [comment]
@tkaemming Fixed the tests.

**Review by tkaemming — APPROVED** [review]
🌄

**Pull request merged**